lib/courseUtil.js

```diff
@@ -0,0 +1,73 @@
+const ERR = require('async-stacktrace');
+const { sqlDb, sqlLoader } = require('@prairielearn/prairielib');
+const child_process = require('child_process');
+
+const config = require('./config');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+function getCommitHash(course, callback) {
+    const gitEnv = process.env;
+    if (config.gitSshCommand != null) {
+        gitEnv.GIT_SSH_COMMAND = config.gitSshCommand;
+    }
+    const spawnOptions = {
+        cwd: course.path,
+        env: gitEnv,
+    };
+    const proc = child_process.spawn('git', ['rev-parse', 'HEAD'], spawnOptions);
```

Comment:
Could use `child_process.exec` to avoid having to do all the event handling ourselves?

Reply:
Absolutely - forgot that that existed :)